lab/dp/problems/custi/main.cpp

```diff
@@ -0,0 +1,135 @@
+// Copyright 2020 Nicolae Mara-Ioana <maraioana9967@gmail.com>
+
+// Custi - Programare dinamica
+// Complexitate: O(n^2)
+
+// https://www.infoarena.ro/problema/custi
+
+#include <bits/stdc++.h>
+
+class Task {
+ public:
+    void solve() {
+        read_input();
+        get_result();
+        print_output();
+    }
+
+ private:
+    int n; // dimensiunea matricei
+    
+    std::vector<std::vector<int>> matrix; // matricea de input
+
+    // fiecare element solution[i] reprezinta numarul de custi
+    // de dimensiune maxima i x i ce se pot forma
+    // fiecare celula se foloseste o singura data ca si colt dreapta jos
+    // a unei custi, aceasta fiind cea de dimesniune maxima
+    std::vector<int> solution;
+
+    // citeste datele de intare din fisier si populeaza matricea
+    void read_input() { 
+        std::cin >> n;
+        matrix = std::vector<std::vector<int>>(n, std::vector<int>(n, 0));
+        solution = std::vector<int>(n + 1, 0);
+        for (int i = 0; i < n; i++) {
+            for (int j = 0; j < n; j++) {
+                std::cin >> matrix[i][j];
+            }
+        }
+    }
+
+    // ideea de la care am plecat: fiecare element este testat daca poate
+    // reprezenta coltul din dreapta jos al unui patrat, astfel:
+    // CAZUL DE BAZA:
+    // dp[i][0] = matrix[i][0]
+    // dp[0][i] = matrix[0][i]
+    // FORMULA: dp[i][j] = min(celelalte 3 colturi) + 1
+    //              (daca elementul poate defini un dreptunghi)
+    //          altfel dp[i][j] = 0
```

Comment:
Let's copy the formula here.

Reply:
What do you mean exactly? Which line?

Reply:
After line 47 I want to see the recurrence